src/server/api/routes/users/v2/follow.js

```diff
@@ -42,12 +43,24 @@ router.post(
         res.status(400).send('INVALID_CREATOR_ADDRESS');
         return;
       }
-      await walletUserCollection.doc(user).update({
-        followers: FieldValue.arrayUnion(creator),
+      await db.runTransaction(async t => {
+        const userRef = walletUserCollection.doc(user);
+        const userDoc = await t.get(userRef);
+        const { email } = userDoc.data();
+        if (!email) throw new Error('EMAIL_NOT_SET_YET');
```

Comment:
Do we need to distinguish better not set and no verified for frontend to take action?